src/animations.jl

```diff
@@ -6,19 +6,55 @@ struct AnimationTrack{T}
     values::Vector{T}
 end
 
+function Base.insert!(track::AnimationTrack, frame::Integer, value)
+    i = searchsortedfirst(track.frames, frame)
+    insert!(track.frames, i, frame)
+    insert!(track.values, i, value)
+    return track
+end
+
+function Base.merge!(a::AnimationTrack, others::AnimationTrack...)
+    for other in others
+        @assert other.name == a.name
+        @assert other.jstype == a.jstype
+        for i in eachindex(other.frames, other.values)
+            # TODO: improve performance.
+            insert!(a, other.frames[i], other.values[i])
+        end
+    end
+    return a
+end
+
 @with_kw struct AnimationClip
     tracks::Dict{String, AnimationTrack} = Dict{String, AnimationTrack}()
     fps::Int = 30
     name::String = "default"
 end
 
+function Base.merge!(a::AnimationClip, others::AnimationClip...)
+    for other in others
+        @assert other.fps == a.fps
+        merge!(merge!, a.tracks, other.tracks) # merge tracks recursively
+    end
+    return a
+end
+
 struct Animation
     clips::Dict{Path, AnimationClip}
     default_framerate::Int
 end
 
 Animation(fps::Int=30) = Animation(Dict{Path, AnimationClip}(), fps)
 
+function Base.merge!(a::Animation, others::Animation...)
+    for other in others
+        # @assert a.default_framerate == other.default_framerate
```

Comment:
I would leave this check in. If the animations don't have matching framerates, then merging them would require figuring out how to interleave their respective frames.

Reply:
The thing that made me hesitate is that it's just the default frame rate. Each of the animation's clips could potentially have a frame rate that's different from the animation's default frame rate. That currently won't happen if you use `atframe` to create the animation, but still.

I'll leave in the assertion, but perhaps something to revisit in the future.